### ISSUE TITLE
F1 score seems wrong when doing binary classification

### ISSUE THREAD

**Issue by antonibigata** [PyTorchLightning/metrics#421]
## 🐛 Bug

When F1 is calculated on a batch where the positive label never appears, the result is 0 even when all predictions are correct.
However, when the labels are positive the result is 1. 
This seems very counterintuitive. Is it expected?

### To Reproduce

import torch
from torchmetrics import F1

f1=F1(num_classes=1)(preds=torch.tensor([0.,0.]), target=torch.tensor([0,0]))
print(f1) # Result is 0

f1=F1(num_classes=1)(preds=torch.tensor([0.8,0.8]), target=torch.tensor([1,1]))
print(f1) # Result is 1

### Expected behavior

I'm not an expert on how F1 is calculated, but I would expect the output to be 1.0 in both cases.

### Environment

pytorch-lightning==1.3.8
torch==1.8.0+cu101
torchmetrics==0.5.0dev

**Comment by SkafteNicki** [comment]
If I run your example using sklearns `f1_score` then we get the same:
```python
from sklearn.metrics import f1_score
f1_score([0,0],[0,0]) # returns 0
f1_score([1,1],[1,1]) # returns 1
```
however, in sklearn you will also get a warning that the first case `f1_score([0,0],[0,0])` is actually ill defined because no true predictions where available. I guess this is the reason why it does not seem to make sense because the score is not really defined in the first case.
We have chosen not give an error in this case, because we do not want users program to crash if they happen to run into this case while training a model.
Closing the issue.

**Comment by antonibigata** [comment]
Thanks a lot for your answer! 
Sklearn does have a parameter "zero_division" that will give the right result if set to 1. Would it be possible to have something similar in torchmetrics?

**Comment by SkafteNicki** [comment]
If you need something right now you can change the default in this line in your local installation:
https://github.com/PyTorchLightning/metrics/blob/5a2388ccaa97cc3608b1fa28879f77436434a6d6/torchmetrics/functional/classification/stat_scores.py#L144
I have to think about this being a parameter exposed to the user :]

**Comment by antonibigata** [comment]
Cheers!

**Comment by e-pet** [comment]
I'd just like to comment that I'd very much like a) a warning to be shown - otherwise it's essentially a silent fail, which as a user I always find particularly frustrating - and b) an option for how to handle zero divisions being exposed to the user.
